### PR TITLE
Mesh peer retry and vault changes.

### PR DESCRIPTION
**Pull request by Tingting-He-ODC**
Ignore the branch name... Mainly adding mesh peer retry and getting mysql password from vault.

### CODE REVIEW

**Comment by Tingting-He-ODC** [comment]
Still need to change mesh release version first.

File: hydra-main/src/main/java/com/addthis/hydra/job/store/DataStoreUtil.java [hunk 1]
```diff
@@ -109,6 +110,11 @@ public static SpawnDataStore makeSpawnDataStore(DataStoreType type) throws Excep
         Properties properties = new Properties();
         if (sqlUser != null) {
             properties.put("user", sqlUser);
+            if(sqlPasspath.length() != 0) {
```
Comment (tedpearson) [inline hydra-main/src/main/java/com/addthis/hydra/job/store/DataStoreUtil.java]: use !isEmpty() :)

File: hydra-main/src/main/java/com/addthis/hydra/job/store/DataStoreUtil.java [hunk 2]
```diff
@@ -109,6 +110,11 @@ public static SpawnDataStore makeSpawnDataStore(DataStoreType type) throws Excep
...
+                VaultWrapper vaultClient = (VaultWrapper) Class.forName("com.addthis.hydra.vault.VaultWrapperImpl").newInstance();//new VaultWrapperImpl();
```
Comment (tedpearson) [inline hydra-main/src/main/java/com/addthis/hydra/job/store/DataStoreUtil.java]: Hard coding a classname that only exists in closed source is not the right way to do this.

File: hydra-main/src/main/java/com/addthis/hydra/job/store/DataStoreUtil.java [hunk 3]
```diff
@@ -109,6 +110,11 @@ public static SpawnDataStore makeSpawnDataStore(DataStoreType type) throws Excep
...
+                vaultClient.login("https://vault.prd.phxshared.oracledatacloud.com");
```
Comment (tedpearson) [inline hydra-main/src/main/java/com/addthis/hydra/job/store/DataStoreUtil.java]: Url should not be hard coded.

**Comment by tedpearson** [comment]
Looks better

**Pull request merged**